src/System.Net.Http/src/Internal/Mail/MailBnfHelper.cs

```diff
@@ -330,9 +333,21 @@ internal static string GetDateTimeString(DateTime value, StringBuilder builder)
                 localBuilder.Append(' ');
             }
 
-            string[] offsetFields = offset.Split(':');
-            localBuilder.Append(offsetFields[0]);
-            localBuilder.Append(offsetFields[1]);
+            int colonIndex = offset.IndexOf(':');
+            if (colonIndex == -1)
+            {
+                throw new InvalidOperationException("TimeSpan.ToString() should contain at least one colon.");
```

Comment:
Where does the input for the method come from?  If the input strings come from the developer/end-user, then this code path needs exceptions.  But if the input is coming from code logic generated internally, then we would use Asserts to validate our algorithm logic.  

This PR changed from using `.Append` to `.IndexOf` methods.  Looking at this, it seems like we still need exceptions. But we can't surface them if they are of a different exception type than before since that would change external behavior.